### ISSUE TITLE
feat: Add kafka binding

### ISSUE THREAD

**Issue by flux-benj** [cloudevents/sdk-ruby#80]
👋 https://github.com/cloudevents/spec/blob/main/cloudevents/bindings/kafka-protocol-binding.md contains a few intricacies which differ from the http specification, making the existing bindings this sdk supports unsuitable for kafka transport. Currently our org extends the http binding and transforms attributes to fit the above specification, however we would be interested in contributing back a specific kafka binding to standardise on approaches. Would you be open to such contribution?

**Comment by dazuma** [comment]
Yes absolutely, it would be great if we had a native kafka binding.
